packages/eth/src/websocket/connection.rs

```diff
@@ -73,15 +74,67 @@ sol!(
 pub struct WsConnection {
     provider: WsProvider,
     address: Address,
-    first_tx_gas_estimation_multiplier: Option<u64>,
-    first_blob_tx_sent: Arc<AtomicBool>,
     blob_provider: Option<WsProvider>,
     blob_signer_address: Option<Address>,
     contract: FuelStateContract,
     commit_interval: NonZeroU32,
+    tx_max_fee: u128,
+    send_tx_request_timeout: Duration,
     metrics: Metrics,
 }
 
+impl WsConnection {
+    async fn get_next_blob_fee(&self, provider: &WsProvider) -> Result<u128> {
+        provider
+            .get_block_by_number(BlockNumberOrTag::Latest, false)
+            .await?
+            .ok_or(Error::Network(
+                "get_block_by_number returned None".to_string(),
+            ))?
+            .header
+            .next_block_blob_fee()
+            .ok_or(Error::Network(
+                "next_block_blob_fee returned None".to_string(),
+            ))
+    }
+
+    async fn get_bumped_fees(
+        &self,
+        previous_tx: &L1Tx,
+        provider: &WsProvider,
+    ) -> Result<(u128, u128, u128)> {
+        let next_blob_fee = self.get_next_blob_fee(provider).await?;
+        let max_fee_per_blob_gas = max(next_blob_fee, previous_tx.blob_fee.saturating_mul(2));
+
+        let Eip1559Estimation {
+            max_fee_per_gas,
+            max_priority_fee_per_gas,
+        } = provider.estimate_eip1559_fees(None).await?;
+
+        let max_fee_per_gas = max(max_fee_per_gas, previous_tx.max_fee.saturating_mul(2));
+        let max_priority_fee_per_gas = max(
+            max_priority_fee_per_gas,
+            previous_tx.priority_fee.saturating_mul(2),
+        );
+
+        Ok((
+            max_fee_per_gas,
+            max_priority_fee_per_gas,
+            max_fee_per_blob_gas,
+        ))
+    }
+
+    fn get_max_fee(tx: &L1Tx, gas_limit: u128, num_fragments: usize) -> u128 {
+        let max_fee = tx.max_fee.saturating_mul(gas_limit).saturating_add(
```

Comment:
This feels off, multiplying a fee with a gas and adding a fee on top of that? Seems like the units don't add up.

Reply:
It's called max_fee in L1Tx for brevity but it actually corresponds to max_fee_per_gas